packages/near-membrane-dom/src/browser-realm.ts

```diff
@@ -185,32 +169,79 @@ export default function createVirtualEnvironment(
         instrumentation,
     });
     linkIntrinsics(env, globalObjectVirtualizationTarget);
-    // window
-    // window.document
-    // In browsers globalThis is === window.
-    if (typeof globalThis === 'undefined') {
-        // Support for globalThis was added in Chrome 71.
-        // However, environments like Android emulators are running Chrome 69.
-        env.link('window', 'document');
+
+    if (SUPPORTS_SHADOW_REALM) {
+        let unsafeBlueDescMap: PropertyDescriptorMap;
+        if (globalObjectVirtualizationTarget === window) {
+            unsafeBlueDescMap = defaultGlobalPropertyDescriptorMap!;
+        } else {
+            unsafeBlueDescMap = { __proto__: null } as unknown as PropertyDescriptorMap;
+            assignFilteredGlobalDescriptorsFromPropertyDescriptorMap(
+                unsafeBlueDescMap,
+                ObjectGetOwnPropertyDescriptors(globalObjectVirtualizationTarget)
+            );
+            for (const key in unsafeBlueDescMap) {
+                if (!(key in defaultGlobalOwnKeysRegistry)) {
+                    delete unsafeBlueDescMap[key];
+                }
+            }
+        }
+        env.link('globalThis');
+        env.redCallableSetPrototypeOf(
+            env.redGlobalThisPointer as Pointer,
+            env.blueGetTransferableValue(blueRefs.WindowProto) as Pointer
+        );
+        env.remap(blueRefs.window, unsafeBlueDescMap);
```

Comment:
☝️ In my attempt even to (temporarily exposing bits to for sure for sure)  set the window prototype. In the realm `window.addEventListener` is `undefined` `window.__proto__` is `{}`. But `Reflect.getPrototypeOf(window).addEventListener` is found. So something is wonky. I tried not linking `globalThis` too and it still biffs. 

I simplified the approach to not be lazy since the realm doesn't have any existing things to bolt on to.